### PR TITLE
Split Math and Interop out of OpenLoco

### PR DESCRIPTION
**Pull request by ZehMatt**

### CODE REVIEW

File: src/Interop/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        ${private_files}
+)
+
+# Disable optimizations for interop.cpp for all compilers, to allow optimized
+# builds without need for -fno-omit-frame-pointer
+if (MSVC)
+    set_source_files_properties("${CMAKE_CURRENT_SOURCE_DIR}/src/Interop/Interop.cpp" PROPERTIES COMPILE_FLAGS "/Oy- /Od")
+else ()
+    set_source_files_properties("${CMAKE_CURRENT_SOURCE_DIR}/src/Interop/Interop.cpp" PROPERTIES COMPILE_FLAGS "-fno-omit-frame-pointer -O0")
+endif ()
```
Comment (duncanspumpkin) [inline src/Interop/CMakeLists.txt:23]: could you add a trailing endline

File: src/OpenLoco/src/Interop/Hooks.cpp [hunk 2]
```diff
@@ -42,6 +41,7 @@
 #include "ViewportManager.h"
 #include "Widget.h"
 #include <OpenLoco/Core/FileSystem.hpp>
+#include <OpenLoco/Interop/Interop.hpp>
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Interop/Hooks.cpp:44]: There are two functions in Hooks.cpp that have their declarations in interop.hpp. You'll need to create a new header in the OpenLoco library or move them to a new header.

registerHooks
loadSections

**Review by duncanspumpkin — APPROVED** [review]
All looks sensible

**Review by AaronVanGeffen — APPROVED** [review]
More modularity sounds like a good way forward. Haven't been able to test these changes, but no objections from me.

**Pull request merged**